### ISSUE TITLE
Address prop spreading warning in WordPress/Gutenberg projects with block API version 2

### ISSUE THREAD

**Issue by ryanwelcher** [10up/eslint-config#32]
With [version 2 of the block AP](https://make.wordpress.org/core/2020/11/18/block-api-version-2/)I, a new hook called `useBlockProps` was introduced to cut down on the amount of extra markup being generated. The [recommended usage](https://developer.wordpress.org/block-editor/developers/block-api/block-edit-save/#block-wrapper-props) for this hook is to spread the returned object on the root of the block.

```jsx
import { useBlockProps } from '@wordpress/block-editor';
 
// ...
const blockSettings = {
    apiVersion: 2,
 
    // ... 
 
    edit: () => {
        const blockProps = useBlockProps( { className: 'my-random-classname' } );
 
        return <div { ...blockProps }>Your block.</div>;
    }
};
```

With the current eslint config for WordPress, this triggers the `react/jsx-props-no-spreading` rule that needs to be disabled either in the file or for each line. This is going to lead to confusion for developers who are getting warnings when they are doing it the way the docs specify.

**Describe the solution you'd like**
Spreading props is not a good thing but is needed in this case and for `userInnerBlockProps` which is available in the the plugin now.  Perhaps we can modify the output of the message or provide an custom formatter with a message that indicates it's OK for these two cases.
